tests/test_contact_wrapper_classes.py

```diff
@@ -3,262 +3,260 @@
-    mock_rod_two = MockRod()
-    mock_rod_two.position_collection = np.array(
-        [
-            [100, 200, 300, 400, 500],
-            [600, 700, 800, 900, 100],
-            [110, 120, 130, 140, 150],
-        ]
-    )
-    ext_contact = ExternalContact(k=1.0, nu=0.5)
-    mock_rod_one_external_forces_before_execution = mock_rod_one.external_forces.copy()
-    mock_rod_two_external_forces_before_execution = mock_rod_two.external_forces.copy()
-    ext_contact.apply_forces(mock_rod_one, 0, mock_rod_two, 0)
-
-    assert_allclose(
-        mock_rod_one.external_forces,
-        mock_rod_one_external_forces_before_execution,
-        rtol=tol,
-        atol=tol,
-    )
-
-    assert_allclose(
-        mock_rod_two.external_forces,
-        mock_rod_two_external_forces_before_execution,
-        rtol=tol,
-        atol=tol,
-    )
-
-
-def test_self_contact_with_rod_self_collision():
-
-    "Testing Self Contact wrapper rod self collision with analytical verified values"
-
-    tol = 1e-6
-    mock_rod = MockRod()
-    sel_contact = SelfContact(k=1.0, nu=0.5)
-    sel_contact.apply_forces(mock_rod, 0, mock_rod, 0)
-
-    assert_allclose(
-        mock_rod.external_forces,
-        np.array(
-            [
-                [-0.976629, -0.663611, -0.934193, -0.471303, -0.028112],
-                [-1.125742, -2.077304, -1.465054, -0.595825, -0.876467],
-                [0.204132, 0.967667, 0.062506, -0.299531, 0.776233],
-            ]
-        ),
-        rtol=tol,
-        atol=tol,
-    )
-
-
-def test_self_contact_with_rod_no_self_collision():
-
-    "Testing Self Contact wrapper rod no self collision with analytical verified values"
-
-    tol = 1e-6
-    mock_rod = MockRod()
-    mock_rod.position_collection = np.array(
-        [[1, 3, 5, 7, 9], [11, 13, 15, 17, 19], [21, 23, 25, 27, 29]]
-    )
-    sel_contact = SelfContact(k=1.0, nu=0.5)
-    sel_contact.apply_forces(mock_rod, 0, mock_rod, 0)
-
-    assert_allclose(
-        mock_rod.external_forces,
-        np.array(
-            [
-                [-0.77306421, -0.25648047, -0.93419262, -0.77665042, -0.33345937],
-                [-1.04834225, -1.92250527, -1.46505411, -0.71192403, -0.99256648],
-                [0.33465609, 1.22871475, 0.06250578, -0.49531749, 0.58044695],
-            ]
-        ),
-        rtol=tol,
-        atol=tol,
+    self.position_collection = np.array([[0], [0], [0]])
+    self.director_collection = np.array(
+        [[[1.0], [0.0], [0.0]], [[0.0], [1.0], [0.0]], [[0.0], [0.0], [1.0]]]
     )
+    self.radius = np.array([1.0])
+    self.length = np.array([2.0])
+    self.external_forces = np.array([[0.0], [0.0], [0.0]])
+    self.external_torques = np.array([[0.0], [0.0], [0.0]])
+    self.velocity_collection = np.array([[0.0], [0.0], [0.0]])
+
+
+MockRod = type("MockRod", (RodBase,), {"__init__": mock_rod_init})
+
+MockRigidBody = type(
+    "MockRigidBody", (RigidBodyBase,), {"__init__": mock_rigid_body_init}
+)
+
+
+class TestExternalContact:
```

Comment:
why did you convert it to class structure?

Reply:
I used classes to couple the test cases that test the same classes, moreover help in code readability
Should I use comments for separation instead?

Reply:
Okay it is fine thanks